stingray/stats.py

```diff
@@ -0,0 +1,268 @@
+
+
+def z2_n_probability(z2, n=2, ntrial=1, n_summed_spectra=1):
+    """Calculate the probability of a certain folded profile, due to noise.
+
+    Parameters
+    ----------
+    z2 : float
+        A Z^2_n statistics value
+    n : int, default 2
+        The ``n`` in $Z^2_n$ (number of harmonics, including the fundamental)
+
+    Other Parameters
+    ----------------
+    ntrial : int
+        The number of trials executed to find this profile
+    n_summed_spectra : int
+        Number of Z_2^n periodograms that were averaged to obtain z2
+
+    Returns
+    -------
+    p : float
+        The probability that the Z^2_n value has been produced by noise
+    """
+    epsilon_1 = stats.chi2.sf(z2 * n_summed_spectra,
+                              2 * n * n_summed_spectra)
+    epsilon = p_multitrial_from_single_trial(epsilon_1, ntrial)
+
+    return epsilon
+
+
+def z2_n_detection_level(n=2, epsilon=0.01, ntrial=1, n_summed_spectra=1):
+    """Return the detection level for the Z^2_n statistics.
+
+    See Buccheri et al. (1983), Bendat and Piersol (1971).
+
+    Parameters
+    ----------
+    n : int, default 2
+        The ``n`` in $Z^2_n$ (number of harmonics, including the fundamental)
+    epsilon : float, default 0.01
+        The fractional probability that the signal has been produced by noise
+
+    Other Parameters
+    ----------------
+    ntrial : int
+        The number of trials executed to find this profile
+    n_summed_spectra : int
+        Number of Z_2^n periodograms that are being averaged
+
+    Returns
+    -------
+    detlev : float
+        The epoch folding statistics corresponding to a probability
+        epsilon * 100 % that the signal has been produced by noise
+    """
+
+    epsilon = p_single_trial_from_p_multitrial(epsilon, ntrial)
+    retlev = stats.chi2.isf(epsilon.astype(np.double),
+                            2 * n_summed_spectra * n) / (n_summed_spectra)
+
+    return retlev
+
+
+def pds_probability(level, ntrial=1, n_summed_spectra=1, n_rebin=1):
+    r"""Give the probability of a given power level in PDS.
+
+    Return the probability of a certain power level in a Power Density
+    Spectrum of nbins bins, normalized a la Leahy (1983), based on
+    the 2-dof :math:`{\chi}^2` statistics, corrected for rebinning (n_rebin)
+    and multiple PDS averaging (n_summed_spectra)
+
+    Parameters
+    ----------
+    level : float or array of floats
+        The power level for which we are calculating the probability
+
+    Other Parameters
+    ----------------
+    ntrial : int
+        The number of *independent* trials (the independent bins of the PDS)
+    n_summed_spectra : int
+        The number of power density spectra that have been averaged to obtain
+        this power level
+    n_rebin : int
+        The number of power density bins that have been averaged to obtain
+        this power level
+
+    Returns
+    -------
+    epsilon : float
```

Comment:
It seems like in other contexts, we've been using `epsilon` to denote the detection level, but here it seems to denote the p-value? Could we change this variable to `p-value` to be more descriptive and not to be confused with the detection level (e.g. the `epsilon` parameter in `pds_detection_level` below)? 

Side note: what we're really computing here (I think/hope?) are p-values (aka a tail probability), rather than any old probability. This might be nitpicky, but I do think there's value in being precise in the documentation about what exactly is computed here.

Reply:
Yes, it's the p-value. 
I use `epsilon`, as a habit, because the Bible (van der Klis etc. etc.) used that :). But it's the same. I will change the name to pvalue